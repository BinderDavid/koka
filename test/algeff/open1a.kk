// test lifting of cps functions
function myid() : ((x:a) -> e a) {
  return fun(x){ x }
}

val hs : forall<e> list<int -> e int> = [id]

function bar(x:int) : <exn> int {
  head(hs)(x)
}
