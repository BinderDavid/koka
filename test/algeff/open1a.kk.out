42
bar : (x : int) -> exn int
hs : forall<e> list<(int) -> e int>
main : () -> <exn,console> ()